cg/apps/avatar/api.py

```diff
@@ -0,0 +1,80 @@
+import random
+from time import sleep
+from typing import Optional
+
+import petname
+import requests
+from bing_image_urls import bing_image_urls
+
+
+class Avatar:
+    @staticmethod
+    def get_avatar_url(internal_id: str) -> Optional[bool]:
+        for url in Avatar.get_avatar_urls(internal_id):
+            if Avatar.is_url_image(url):
+                return url
+
+    @staticmethod
+    def get_avatar_urls(internal_id: str) -> Optional[bool]:
+
+        urls = None
+        adjective, animal = Avatar._split_petname(internal_id)
+        filter_array = [
+            "+filterui:license-L2_L3",
+            "+filterui:aspect-square",
+            "+filterui:photo-transparent",
+            "+filterui:imagesize-small",
+        ]
+
+        try_cnt = 0
+        filters = "".join(filter_array)
+        while not urls and filters:
+            query = f"{adjective} {animal}"
+            urls = bing_image_urls(
+                query=query,
+                filters=filters,
+                verify_status_only=False,
+            )
+            sleep(random.randint(0, len(filter_array) - try_cnt))
+            if not urls:
+                query = f"{animal}"
+                urls = bing_image_urls(
+                    query=query,
+                    filters=filters,
+                    verify_status_only=False,
+                )
+
+            try_cnt += 1
+            filters = "".join(filter_array[: len(filter_array) - try_cnt])
+
+        random.shuffle(urls)
+        return urls
+
+    @staticmethod
+    def is_url_image(image_url: str) -> bool:
+        from PIL import Image
+
+        try:
+            img = Image.open(requests.get(image_url, stream=True).raw)
+            width, height = img.size
+
+            if 0 < width and 0 < height:
+                return True
+        except Exception:
+            return False
+
+        return False
+
+    @classmethod
+    def _split_petname(cls, merged_name: str) -> (str, str):
+
+        found_adj = ""
+        for adjective in petname.adjectives:
+            if adjective in merged_name:
+                found_adj = adjective
+        found_pet = ""
+        for pet in petname.names:
+            if pet in merged_name:
+                found_pet = pet
+
+        return found_adj, found_pet
```

Comment:
What happens in the case of partial pet name matches? For instance, you check for `cat`, but the actual pet name is `caterpillar`?

Reply:
really good catch :) do you see any way we can fix it?

Reply:
maybe just search for `adjective`, once one is found strip it from `merged_name`, whatever remains is `found_pet`?

Reply:
I fixed it by sorting the lists so we get the longest matches first. There might we some edge cases overlooked but I think it will be a lot better than before :)